app/controllers/DataRetrievals.scala

```diff
@@ -70,4 +76,106 @@ object DataRetrievals {
     }
   }
 
+  def cyaChargeGeneric[A](chargeDetailsPage: QuestionPage[A],
+                          srn: String)
+                         (block: (A, String) => Future[Result])
+                         (implicit request: DataRequest[AnyContent], reads: Reads[A]): Future[Result] = {
+    (
+      request.userAnswers.get(chargeDetailsPage),
+      request.userAnswers.get(SchemeNameQuery)
+    ) match {
+      case (Some(chargeDetails), Some(schemeName)) =>
+        block(chargeDetails, schemeName)
+      case _ =>
+        Future.successful(Redirect(controllers.routes.AFTSummaryController.onPageLoad(srn, None)))
+    }
+  }
+
+  def cyaChargeC(isSponsoringEmployerIndividualPage: QuestionPage[Boolean],
```

Comment:
Do we need to bother passing these in? They will never be anything other than these values, will they?

Reply:
Could just pass in the index

Reply:
Done